lms/djangoapps/courseware/views/views.py

```diff
@@ -707,6 +708,63 @@ def course_about(request, course_id):
         return render_to_response('courseware/course_about.html', context)
 
 
+# TODO: make this efficient
+def _add_course_info(program, user):
+    for program_course in program['courses']:
+        for course_run in program_course['course_runs']:
+            course = modulestore().get_course(CourseKey.from_string(course_run['key']))
+            course_run.update({
+                'registered': registered_for_course(course, user),
+                'can_enroll': bool(has_access(user, 'enroll', course)),
+                'is_shib_course': uses_shib(course)
+            })
+
+
+def _get_youtube_url_identifier(url):
```

Comment:
I'm still not understanding why we need this. Wouldn't we just configure the video URL we need in the catalog service's Django admin when setting up the program?